compiler-core/src/language_server/tests/completion.rs

```diff
@@ -726,3 +726,71 @@ pub fn wibble(
         .concat()
     );
 }
+
+#[test]
+fn public_internal_definitions_from_outer_modules_are_not_in_the_completions() {
+    let code = r#"
+@external(erlang, "rand", "uniform")
+@internal
+pub fn random_float() -> Float
+
+@internal pub fn main() { 0 }
+@internal pub type Alias = Int
+@internal pub type Foo { Bar }
+@internal pub const foo = 1
+"#;
+
+    assert_eq!(expression_completions("", code), vec![]);
+}
+
+#[test]
+fn public_internal_definitions_from_the_same_module_are_in_the_completions() {
```

Comment:
This test seems to only check value-completion, not type-completions. In particular the types `Foo` and `Alias`.

I think these would need a separate test (like `for_custom_type_definition()` at line 568 or the five following tests) which would then detect the I-think-incorrect behaviour in my above comment.